src/hotspot/cpu/arm/c1_LIRAssembler_arm.cpp

```diff
@@ -2432,6 +2432,11 @@ void LIR_Assembler::emit_lock(LIR_OpLock* op) {
   Register lock = op->lock_opr()->as_pointer_register();
 
   if (UseHeavyMonitors) {
+    if (op->info() != NULL) {
+      int null_check_offset = __ offset();
+      __ null_check(obj);
+      add_debug_info_for_null_check(null_check_offset, op->info());
```

Comment:
Is this equivalent to the following?
```
add_debug_info_for_null_check_here(op->info());
__ null_check(obj);
```

Reply:
I don't know.  Is it better and preferable?

Reply:
I like it better.  Thanks for changing it.